src/formatDate.js

```diff
@@ -8,7 +8,51 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateParts = date.split(/[-/.]/);
```

Comment:
The use of a regular expression to split the date parts is clever, but it assumes that the input format's separator will always be one of the characters in the set [-/.]. If the format were to change to include a new separator, this line would fail to split the date correctly. Consider using the `fromFormat` array to determine the separator instead.